### PR TITLE
feat: add generic action to task

### PR DESCRIPTION
**Pull request by lstocchi**
### What does this PR do?

It adds a new property to a task to implement a custom behavior like show an 'open folder' action.
This will be used with the export container action. When the export is executing/has been executed the user could open the taskmanager, click on the open folder link to open the directory where the container content will be saved

### Screenshot / video of UI

N/A

### What issues does this PR fix or reference?

it is part of #6355 

### How to test this PR?

1. run test

- [x] Tests are covering the bug fix or the new feature

### CODE REVIEW

File: packages/extension-api/src/extension-api.d.ts [hunk 1]
```diff
@@ -1500,7 +1500,7 @@ declare module '@podman-desktop/api' {
     /**
      * The resource the dialog shows when opened.
      */
-    defaultUri?: Uri;
+    defaultUri?: Uri | string;
```
Comment (jeffmaury) [inline packages/extension-api/src/extension-api.d.ts]: Can't we keep it as Uri only and assume that it is a file uri ?

Comment (lstocchi) [reply packages/extension-api/src/extension-api.d.ts]: What do you mean exactly? Bc i would have to call the openDialog from the client. So i would have something like `window.openDialog(defaultUri: mypath)`. I could create a fake Uri myself with 
```
scheme: file,
path: mypath,
fspath: mypath, 
....
```
but it's a waste imo. In the openDialog it's going to take the fsPath in any case

Comment (jeffmaury) [reply packages/extension-api/src/extension-api.d.ts]: I meant the API is already designed to open a file or folder so why do you have to update the API ?

Comment (lstocchi) [reply packages/extension-api/src/extension-api.d.ts]: bc there was no way to create a URI item from the frontend and bc the defaultUri can be basically be a string i was extending it. I'm going to create another PR to expose the createUri function and revert it

Comment (lstocchi) [reply packages/extension-api/src/extension-api.d.ts]: opened https://github.com/containers/podman-desktop/pull/6454 . I'll use that when calling the openDialog

File: packages/renderer/src/lib/task-manager/task-manager.ts [hunk 2]
```diff
@@ -40,6 +40,7 @@ export class TaskManager {
         status: task.status,
         hasGotoTask: false,
         age: `${humanizeDuration(new Date().getTime() - task.started, { round: true, largest: 1 })} ago`,
+        openFolder: task.openFolder,
```
Comment (benoitf) [inline packages/renderer/src/lib/task-manager/task-manager.ts]: maybe a test in task-manager.spec.ts ?

File: packages/renderer/src/lib/task-manager/TaskManagerItem.svelte [hunk 3]
```diff
@@ -111,6 +115,21 @@ function gotoTask(taskUI: StatefulTaskUI) {
...
+    {#if isStatefulTask(taskUI) && taskUI.status !== 'failure'}
+      <div class="flex flex-row w-full">
+        <div class="flex flex-1 flex-col w-full items-end text-purple-500 text-xs">
+          {#if taskUI.openFolder}
+            <button
+              class="text-purple-500 cursor-pointer"
+              on:click="{() => {
+                if (isStatefulTask(taskUI)) openFolder(taskUI);
+              }}"
+              aria-label="open folder">Open folder ></button>
```
Comment (benoitf) [inline packages/renderer/src/lib/task-manager/TaskManagerItem.svelte]:
```suggestion
              aria-label="open folder">Open folder ></button>
```

is that the `>` after Open Folder is wanted ?

Comment (lstocchi) [reply packages/renderer/src/lib/task-manager/TaskManagerItem.svelte]: I did that on purpose to follow the open task but it can be removed.
open_folder

Comment (benoitf) [reply packages/renderer/src/lib/task-manager/TaskManagerItem.svelte]: maybe use `&gt;` so we know it's an html entity ?

File: packages/renderer/src/lib/task-manager/task-manager.ts (hunk 2 above)
Comment (lstocchi) [reply packages/renderer/src/lib/task-manager/task-manager.ts]: Done

File: packages/renderer/src/lib/task-manager/TaskManagerItem.svelte (hunk 3 above)
Comment (lstocchi) [reply packages/renderer/src/lib/task-manager/TaskManagerItem.svelte]: done

**Review by benoitf — APPROVED** [review]

**Comment by deboer-tim** [comment]
I'm finding it hard to follow the comments and changes afterwards, but if this went from a 'folder URI' to a function -which could do anything - why keep the property named 'folder'? Wouldn't it make sense to allow things like 'open file' or 'open logs' as well? Maybe even 'open browser'? Those seem just as likely for other tasks, and would only require a label as well.

**Comment by lstocchi** [comment]
> I'm finding it hard to follow the comments and changes afterwards, but if this went from a 'folder URI' to a function -which could do anything - why keep the property named 'folder'? Wouldn't it make sense to allow things like 'open file' or 'open logs' as well? Maybe even 'open browser'? Those seem just as likely for other tasks, and would only require a label as well.

It was always a function, the thing discussed was how to pass the info ( as a string or as a Uri) to the backend. 
BTW yes, it could be used for anything. But the same for the goToTask.
So you would suggest to have something generic like` action: { title, function }`, so that the title is the text displayed and function is the one executed when clicking?

**Comment by deboer-tim** [comment]
> > I'm finding it hard to follow the comments and changes afterwards, but if this went from a 'folder URI' to a function -which could do anything - why keep the property named 'folder'? Wouldn't it make sense to allow things like 'open file' or 'open logs' as well? Maybe even 'open browser'? Those seem just as likely for other tasks, and would only require a label as well.
> 
> It was always a function, the thing discussed was how to pass the info ( as a string or as a Uri) to the backend. BTW yes, it could be used for anything. But the same for the goToTask. So you would suggest to have something generic like` action: { title, function }`, so that the title is the text displayed and function is the one executed when clicking?

Yes, I think that's what I'm suggesting. Maybe it is even worth replacing goToTask (not necessarily in this PR...) or allowing more than one. We should keep some consistency in the labels, but this gives any task a simple and generic way to link to relevant things, whether it's a status page, logs, a folder, help, etc.

**Comment by lstocchi** [comment]
@deboer-tim updated 👍

**Review by benoitf — APPROVED** [review]

**Review by deboer-tim — APPROVED** [review]

**Comment by deboer-tim** [comment]
Thanks for the change. 👍🏼

**Review by jeffmaury — APPROVED** [review]
LGTM
